src/fparser/two/tests/fortran2008/test_open_stmt_r904.py

```diff
@@ -0,0 +1,146 @@
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+"""Test Fortran 2008 rule R904
+
+   open-stmt is OPEN ( connect-spec-list )
+
+   R905 connect-spec is [ UNIT = ] file-unit-number
+                    or ACCESS = scalar-default-char-expr
+                    or ACTION = scalar-default-char-expr
+                    or ASYNCHRONOUS = scalar-default-char-expr
+                    or BLANK = scalar-default-char-expr
+                    or DECIMAL = scalar-default-char-expr
+                    or DELIM = scalar-default-char-expr
+                    or ENCODING = scalar-default-char-expr
+                    or ERR = label
+                    or FILE = file-name-expr
+                    or FORM = scalar-default-char-expr
+                    or IOMSG = iomsg-variable
+                    or IOSTAT = scalar-int-variable
+                    or NEWUNIT = scalar-int-variable
+                    or PAD = scalar-default-char-expr
+                    or POSITION = scalar-default-char-expr
+                    or RECL = scalar-int-expr
+                    or ROUND = scalar-default-char-expr
+                    or SIGN = scalar-default-char-expr
+                    or STATUS = scalar-default-char-expr
+
+R906 file-name-expr is scalar-default-char-expr
+R907 iomsg-variable is scalar-default-char-variable
+
+C903 No specifier shall appear more than once in a given connect-spec-list.
+C904 (R904) If the NEWUNIT= specifier does not appear, a file-unit-number
+     shall be specified; if the optional characters UNIT= are omitted, the
+     file-unit-number shall be the first item in the connect-spec-list.
+C905 (R904) The label used in the ERR= specifier shall be the statement label
+     of a branch target statement that appears in the same inclusive scope as
+     the OPEN statement.
+C906 (R904) If a NEWUNIT= specifier appears, a file-unit-number shall not
+     appear.
+
+"""
+
+import pytest
+from fparser.api import get_reader
+from fparser.two import Fortran2008
+from fparser.two.utils import NoMatchError, walk
+
+
+@pytest.mark.parametrize(
+    "open_args",
+    [
+        "unit=23, file='hello'",
+        "23, file='hello', err=36",
+        "23, file='hello', status='OLD'",
+        "unit=23, file='hello', action='read'",
+    ],
+)
+def test_open_f2003_args(f2008_parser, open_args):
```

Comment:
You can use the @pytest.mark.usefixtures("f2003_create") to avoid having an unused argument

Reply:
Done throughout this file.